vavr/src/test/java/com/baeldung/vavr/collections/CollectionAPIUnitTest.java

```diff
@@ -0,0 +1,261 @@
+        assertEquals(new Integer(2), intVector.get(1));
+        assertEquals(new Integer(3), intVector.get(2));
+        assertEquals(new Integer(6), newVector.get(1));
+    }
+
+    @Test
+    public void givenCharSeq_whenProcessed_thenCorrect() {
+        CharSeq chars = CharSeq.of("vavr");
+        CharSeq newChars = chars.replace('v', 'V');
+
+        assertEquals(4, chars.size());
+        assertEquals(4, newChars.size());
+
+        assertEquals('v', chars.charAt(0));
+        assertEquals('V', newChars.charAt(0));
+        assertEquals("Vavr", newChars.mkString());
+    }
+
+    @Test
+    public void givenHashSet_whenModified_thenCorrect() {
+        HashSet<String> set = HashSet.of("Red", "Green", "Blue");
+        HashSet<String> newSet = set.add("Yellow");
+
+        assertEquals(3, set.size());
+        assertEquals(4, newSet.size());
+        assertFalse(set.contains("Yellow"));
+        assertTrue(newSet.contains("Yellow"));
+    }
+
+    @Test
+    public void givenSortedSet_whenIterated_thenCorrect() {
+        SortedSet<String> set = TreeSet.of("Red", "Green", "Blue");
+
+        Iterator<String> iterator = set.iterator();
+        assertEquals("Blue", iterator.next());
+        assertEquals("Green", iterator.next());
+        assertEquals("Red", iterator.next());
+    }
+
+    @Test
+    public void givenSortedSet_whenReversed_thenCorrect() {
+        Comparator<String> reverseCompare = (a, b) -> b.compareTo(a);
+        SortedSet<String> set = TreeSet.of(reverseCompare, "Green", "Red", "Blue");
+
+        Iterator<String> iterator = set.iterator();
+        assertEquals("Red", iterator.next());
+        assertEquals("Green", iterator.next());
+        assertEquals("Blue", iterator.next());
+    }
+
+    @Test
+    public void givenMap_whenIterated_thenCorrect() {
+        Map<Integer, List<Integer>> map = List.rangeClosed(0, 10)
+            .groupBy(i -> i % 2);
+
+        assertEquals(2, map.size());
+
+        Iterator<Tuple2<Integer, List<Integer>>> iterator = map.iterator();
+        assertEquals(6, iterator.next()
+            ._2()
+            .size());
+        assertEquals(5, iterator.next()
+            ._2()
+            .size());
+    }
+
+    @Test
+    public void givenTreeMap_whenIterated_thenCorrect() {
+        SortedMap<Integer, String> map = TreeMap.of(3, "Three", 2, "Two", 4, "Four", 1, "One");
+
+        Iterator<Tuple2<Integer, String>> iterator = map.iterator();
+        assertEquals(new Integer(1), iterator.next()
+            ._1());
+        assertEquals(new Integer(2), iterator.next()
+            ._1());
+        assertEquals(new Integer(3), iterator.next()
+            ._1());
+    }
+
+    @Test
+    public void givenJavaList_whenConverted_thenCorrect() {
+        java.util.List<Integer> javaList = java.util.Arrays.asList(1, 2, 3, 4);
+        List<Integer> vavrList = List.ofAll(javaList);
+
+        assertTrue(vavrList instanceof io.vavr.collection.List);
```

Comment:
Is it Ok to have this example in a private method in the same test class? Just wondering if a separate main class is required just for a few lines of code.

For rest of the comments, will work on it as suggested.